letsencrypt/auth_handler.py

```diff
@@ -113,6 +113,18 @@ def _solve_challenges(self):
                     cont_resp = self.cont_auth.perform(self.cont_c)
                 if self.dv_c:
                     dv_resp = self.dv_auth.perform(self.dv_c)
+                    for achall, response in zip(self.dv_c, dv_resp):
+                        if isinstance(response, challenges.HTTP01Response):
+                            if not response.simple_verify(
+                                achall.chall, achall.domain,
+                                achall.account_key.public_key(), response.PORT):
+                                logger.warning("Self-verify of challenge failed.")
+                        elif isinstance(response, challenges.TLSSNI01Response):
+                            if not response.simple_verify(
+                                achall.chall, achall.domain,
+                                achall.account_key.public_key()):
+                                logger.warning("Self-verify of challenge failed.")
```

Comment:
Please include some kind of challenge ID in the log message - otherwise it will be hard to guess which challenge actually failed if there are multiple being solved.

Same for `http-01`. Though this logging line could be deduplicated, instead of copying it over between all difference subclass branches.